### PR TITLE
Nanoseconds time precision for local files structure cache

### PR DESCRIPTION
**Pull request by azat**
### Checklist
- [ ] use modification time instead of current time for initial cache entry

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Nanoseconds time precision for local files structure cache

Refs: https://github.com/ClickHouse/ClickHouse/pull/59874

### CODE REVIEW

File: base/base/TimeSpec.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <tuple>
+#include <sys/time.h>
+
+struct TimeSpec
```
Comment (vdimir) [inline base/base/TimeSpec.h:6]:
```suggestion
/// Wrapper class for the `timespec` structure that provides easy construction from `time_t` and enables comparison operations.
struct TimeSpec
```

File: src/Storages/System/StorageSystemSchemaInferenceCache.cpp [hunk 2]
```diff
@@ -59,7 +59,7 @@ static void fillDataImpl(MutableColumns & res_columns, SchemaCache & schema_cach
         res_columns[1]->insert(key.source);
         res_columns[2]->insert(key.format);
         res_columns[3]->insert(key.additional_format_info);
-        res_columns[4]->insert(schema_info.registration_time);
+        res_columns[4]->insert(schema_info.registration_time.getTimeSpec().tv_sec);
```
Comment (vdimir) [inline src/Storages/System/StorageSystemSchemaInferenceCache.cpp:62]: Shouldn't we add column with additional precision we have now?

**Review by vdimir — COMMENTED** [review]
Do we really need such a precision there, maybe there's other way to fix an issue?
Cache is used not trigger update less often, but nanosecond precision sounds like it may be changed often.

**Comment by azat** [comment]
So, @vdimir do you think it does not worth it?

**Comment by vdimir** [comment]
Is it true that if a file is changed twice per second, we will miss the second change and continue using the schema inferred for the first version until the file is modified again?
If it's the only problem and fix is quite complicated, I suppose we an give up

**Comment by azat** [comment]
>Is it true that if a file is changed twice per second, we will miss the second change and continue using the schema inferred for the first version until the file is modified again?

Yes.

**Pull request closed without merging**